OpenSim/Analyses/OutputReporter.cpp

```diff
@@ -58,31 +61,30 @@ int OutputReporter::begin(const SimTK::State& s)
     _minimumStageToRealize = SimTK::Stage::Time;
 
     for (int i = 0; i < getProperty_output_paths().size(); ++i) {
-        auto outpath = ComponentPath(get_output_paths(i));
-        // assume that Outputs are for the Model
-        string compName = _pvtModel->getAbsolutePathString();
-        // if a a ComponentPath is specified then use the parent path
-        if (outpath.getNumPathLevels()) {
-            compName = outpath.getParentPathString();
+        std::string componentPath;
+        std::string outputName;
+        std::string channelName;
+        std::string alias;
+        AbstractInput::parseConnecteePath(get_output_paths(i),
+                                          componentPath, outputName,
+                                          channelName, alias);
+        // The user supplied a path relative to the model, but that won't be a
+        // valid relative path from the reporters. Form an absolue path.
+        if (componentPath.empty() || componentPath[0] != '/') {
+            componentPath = "/" + componentPath;
         }
-        auto outputName = outpath.getComponentName();
-
-        // In keeping with assumption of Outputs for the Model (Component)
-        const Component* comp = _pvtModel.get();
-
-        // If the ComponentPath has a Component name get that Component
-        if (!compName.empty())
-            comp = &_pvtModel->getComponent(compName);
-
-        auto& out = comp->getOutput(outputName);
+        const auto path = AbstractInput::composeConnecteePath(
+                componentPath, outputName, channelName, alias);
+        const auto& comp = _pvtModel->getComponent(componentPath);
+        auto& out = comp.getOutput(outputName);
         if (out.getTypeName() == "double") {
-            _tableReporterDouble->addToReport(out);
+            inputDouble.appendConnecteePath(path);
```

Comment:
Might look cleaner as a `switch`.

Reply:
Can't use strings in switch statements; https://stackoverflow.com/questions/650162/why-the-switch-statement-cannot-be-applied-on-strings

Reply:
Ah ok, sorry, wrong language.